### PR TITLE
Refactor unsupported db error messages

### PR DESCRIPTION
**Pull request by AdRiley**
### Pull Request Description

Refactor of the DB Errors before I add a load more of them for my SQL Server feature flag work.

Now the error text is in one spot and shorter.

Before:

image

After:

image

Happy to take feedback on what the error message should say. There were multiple versions prior to this change.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

**Review by marthasharkey — APPROVED** [review]

File: distribution/lib/Standard/Database/0.0.0-dev/src/DB_Column.enso [hunk 1]
```diff
@@ -327,13 +327,13 @@ type DB_Column
...
         checked_support = if self.connection.dialect.is_supported op_kind then True else
-            Error.throw (Unsupported_Database_Operation.Error "The operation "+op_kind+" is not supported by this backend.")
+            Error.throw (Unsupported_Database_Operation.Error op_kind)
         checked_support.if_not_error <|
             type_mapping = self.connection.dialect.get_type_mapping
             prepare_operand operand = case operand of
                 other_column : DB_Column ->
                     if Helpers.check_integrity self other_column then other_column.expression else
-                        Error.throw <| Unsupported_Database_Operation.Error "Cannot use columns coming from different contexts in one expression without a join."
+                        Error.throw <| Error "Cannot use columns coming from different contexts in one expression without a join."
```
Comment (GregoryTravis) [inline distribution/lib/Standard/Database/0.0.0-dev/src/DB_Column.enso]: What does the bare `Error` mean here?

Comment (AdRiley) [reply distribution/lib/Standard/Database/0.0.0-dev/src/DB_Column.enso]: It means I missed replacing it with the correct error type. And that we don't have a test for this line of code :)

Comment (AdRiley) [reply distribution/lib/Standard/Database/0.0.0-dev/src/DB_Column.enso]: Good catch thank you

Comment (AdRiley) [reply distribution/lib/Standard/Database/0.0.0-dev/src/DB_Column.enso]: I have fixed this line and added a ticket to add the tests https://github.com/enso-org/enso/issues/11176

**Review by GregoryTravis — APPROVED** [review]

File: distribution/lib/Standard/Database/0.0.0-dev/src/Errors.enso [hunk 2]
```diff
@@ -6,19 +6,63 @@ polyglot java import java.sql.SQLException
...
-    Error message
+    Error operation:Text
 
     ## PRIVATE
        Convert the SQL error to a textual representation.
-    to_text : Text
-    to_text self =
-        "Unsupported database operation: " + self.message
+    to_text self -> Text =
+        self.operation + " not yet supported for DB backend. Use .read to download the data and use this feature."
```
Comment (radeusgd) [inline distribution/lib/Standard/Database/0.0.0-dev/src/Errors.enso:14]: I'd consider
```suggestion
        self.operation + " not yet supported for DB backend. Use .read to download the data and use this feature in-memory."
```

**Pull request merged**